### PR TITLE
[PyTorch] Add unit test for c10::Synchronized<T>

### PR DESCRIPTION
**Pull request by dhruvbird**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #74062
* #74061
* #74060
* #73725
* #73724
* #73723

Since this class was added, it's missing a unit test. @dbort first noticed it, so adding one in this commit.

Differential Revision: [D34800969](https://our.internmc.facebook.com/intern/diff/D34800969/)

### CODE REVIEW

File: c10/test/util/Synchronized_test.cpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#include "c10/util/Synchronized.h"
+#include <c10/util/Synchronized.h>
```
Comment (dbort) [inline c10/test/util/Synchronized_test.cpp:2]: Duplicate includes

**Review by dbort — COMMENTED** [review]
Thank you so much for adding this! Looks great.

Please take a look at the clang-tidy and clang-format warnings/errors on internal CI.

File: c10/test/util/Synchronized_test.cpp (hunk 1 above)
Comment (dhruvbird) [reply c10/test/util/Synchronized_test.cpp:2]: Thanks VSCode!

**Review by dbort — APPROVED** [review]

**Comment by malfet** [comment]
Why this change was landed when it clearly broke Windows builds and clang-format?

**Pull request closed without merging**